In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Import homelessness dataset and explore it using .head(), .info(), .shape, .describe(), .values, .columns, .index and based on your understanding, write some explanations about each of it

### load homelessness dataset

In [3]:
homelessness = pd.read_csv('../homelessness.csv')

with the use of pandas read_csv method, we have loaded the dataset of homelessness into homelessness variable

### use .head()

In [4]:
homelessness.head()

Unnamed: 0              region       state  individuals  family_members  \
0           0  East South Central     Alabama       2570.0           864.0   
1           1             Pacific      Alaska       1434.0           582.0   
2           2            Mountain     Arizona       7259.0          2606.0   
3           3  West South Central    Arkansas       2280.0           432.0   
4           4             Pacific  California     109008.0         20964.0   

   state_pop  
0    4887681  
1     735139  
2    7158024  
3    3009733  
4   39461588

 by using .head() on homelessness dataset, we get first 5 rows of the dataset for basic understanding, here we can see 
 1. there are 2 index labels, on which we need to remove the unnamed column of index,
 2. we see 5 columns and getting basic understanding of the dataset where
 1. region column states region of the state
 2. state column contains name of the state
 3. individuals column has the numerical value which states how many families which do not have their dependants
 4. family_members column has numerical value which states number of families which are homeless
 5. state_pop column states total population of the state

remove extra index

In [8]:
homelessness = pd.read_csv('../homelessness.csv', index_col=0)

### .info()

In [9]:
homelessness.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 51 entries, 0 to 50
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   region          51 non-null     object 
 1   state           51 non-null     object 
 2   individuals     51 non-null     float64
 3   family_members  51 non-null     float64
 4   state_pop       51 non-null     int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 2.4+ KB


here we can see it cotains 
1. total of 51 rows and 6 columns,
2. data type of each column
3. also there are no missing values present in the dataset
4. individual and family_members column are floating point data type
5. state_pop data type is integer for the number of people in that state

### .shape

In [10]:
homelessness.shape

(51, 5)

above simply tells us the 
1. number of rows: 51
2. number of columns: 5

### .describe()

In [11]:
homelessness.describe()

individuals  family_members     state_pop
count      51.000000       51.000000  5.100000e+01
mean     7225.784314     3504.882353  6.405637e+06
std     15991.025083     7805.411811  7.327258e+06
min       434.000000       75.000000  5.776010e+05
25%      1446.500000      592.000000  1.777414e+06
50%      3082.000000     1482.000000  4.461153e+06
75%      6781.500000     3196.000000  7.340946e+06
max    109008.000000    52070.000000  3.946159e+07

we can see some statistics of numerical columns
1. total count of each numerical cols is 51
2. mean of individuals is 7225, family members is 3504, pop is 6 millions
3. standard deviation
4. minimum individual is 434, family_member is 75, pop is 5.7 millions
5. percentiles
6. maximum

### .values

In [12]:
homelessness.values

array([['East South Central', 'Alabama', 2570.0, 864.0, 4887681],
       ['Pacific', 'Alaska', 1434.0, 582.0, 735139],
       ['Mountain', 'Arizona', 7259.0, 2606.0, 7158024],
       ['West South Central', 'Arkansas', 2280.0, 432.0, 3009733],
       ['Pacific', 'California', 109008.0, 20964.0, 39461588],
       ['Mountain', 'Colorado', 7607.0, 3250.0, 5691287],
       ['New England', 'Connecticut', 2280.0, 1696.0, 3571520],
       ['South Atlantic', 'Delaware', 708.0, 374.0, 965479],
       ['South Atlantic', 'District of Columbia', 3770.0, 3134.0, 701547],
       ['South Atlantic', 'Florida', 21443.0, 9587.0, 21244317],
       ['South Atlantic', 'Georgia', 6943.0, 2556.0, 10511131],
       ['Pacific', 'Hawaii', 4131.0, 2399.0, 1420593],
       ['Mountain', 'Idaho', 1297.0, 715.0, 1750536],
       ['East North Central', 'Illinois', 6752.0, 3891.0, 12723071],
       ['East North Central', 'Indiana', 3776.0, 1482.0, 6695497],
       ['West North Central', 'Iowa', 1711.0, 1038.0, 3148618]

here we can see the data of each rows represented in 2D numpy array

### .columns

In [13]:
homelessness.columns

Index(['region', 'state', 'individuals', 'family_members', 'state_pop'], dtype='object')

we can see the each column names in the dataset

### .index

In [14]:
homelessness.index

Int64Index([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
            17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
            34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49,
            50],
           dtype='int64')

we can see the n